uberfire-m2repo-editor/uberfire-m2repo-editor-backend/src/main/java/org/guvnor/m2repo/backend/server/M2ServletContextListener.java

```diff
@@ -0,0 +1,151 @@
+        logger.info("M2ServletContextListener contextInitialized deployed {} jars in {} sec ", jarsDeployed, totalTime);
+    }
+
+    @Override
+    public void contextDestroyed(ServletContextEvent servletContextEvent) {}
+
+    private String getJarsFolder(){
+        StringBuilder sb = new StringBuilder();
+        sb.append(File.separator).append(WEB_INF_FOLDER).append(File.separator).append(LIB_FOLDER).append(File.separator);
+        return sb.toString();
+    }
+
+    private int deployJarsFromWar(String path) {
+        int i = 0;
+        try (DirectoryStream<Path> ds = Files.newDirectoryStream(Paths.get(path))) {
+            for (Path p : ds) {
+                if (p.toString().endsWith(JAR_EXT)) {
+                    deployJar(p.toAbsolutePath().toString());
+                    i++;
+                }
+            }
+        } catch (Exception e) {
+            logger.error(e.getMessage(), e);
+        }
+        return i;
+    }
+
+    public GAV deployJar(String file) {
+        GAV gav = new GAV();
+        Properties props = readZipFile(file);
+        if(!props.isEmpty()){
+            gav = new GAV(props.getProperty(GROUP_ID),props.getProperty(ARTIFACT_ID), props.getProperty(VERSION) );
+            deploy(gav, file);
+        }
+        return gav;
+    }
+
+
+    public Properties readZipFile(String zipFilePath) {
+        try {
+            ZipFile zipFile = new ZipFile(zipFilePath);
+            Enumeration<? extends ZipEntry> e = zipFile.entries();
+            while (e.hasMoreElements()) {
+                ZipEntry entry = e.nextElement();
+                String begin = "META-INF"+ File.separator + "maven";
+                String end = "pom.properties";
+                if (!entry.isDirectory() && entry.getName().startsWith(begin) && entry.getName().endsWith(end)) {
+                    BufferedInputStream bis = new BufferedInputStream(zipFile.getInputStream(entry));
+                    Properties props = new Properties();
+                    props.load(bis);
+                    bis.close();
+                    return props;
+                } else {
+                    continue;
+                }
+            }
+        } catch (IOException e) {
+            logger.error("IOError :{}",e.getMessage(), e);
+        }
+        return new Properties();
+    }
+
+    public boolean  deploy(GAV gav, String jarFile) {
+        Artifact jarArtifact = new DefaultArtifact(gav.getGroupId(),
+                                                   gav.getArtifactId(),
+                                                   JAR_ARTIFACT,
+                                                   gav.getVersion());
+        jarArtifact = jarArtifact.setFile(new File(jarFile));
+        try {
+            final InstallRequest installRequest = new InstallRequest();
+            installRequest.addArtifact(jarArtifact);
+            InstallResult result = Aether.getAether().getSystem().install(Aether.getAether().getSession(), installRequest);
```

Comment:
are you sure that Aether is pointing to our global m2 repo?

Reply:
It use the {user.home}/.m2/repository of type enhanced, without configure other repos system this is the default

Reply:
so we don't want this... we want it inside our global repo